### PR TITLE
[SPARK-30551][SQL] Disable comparison for interval type

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

As we are not going to follow ANSI to implement year-month and day-time interval types, it is weird to compare the year-month part to the day-time part for our current implementation of interval type now. 

Additionally, the current ordering logic comes from PostgreSQL where the implementation of the interval is messy. And we are not aiming PostgreSQL compliance at all.

THIS PR will revert https://github.com/apache/spark/pull/26681 and https://github.com/apache/spark/pull/26337

### Why are the changes needed?

make interval type more future-proofing

### Does this PR introduce any user-facing change?

there are new in 3.0, so no

### How was this patch tested?

existing uts shall work

### CODE REVIEW

File: common/unsafe/src/main/java/org/apache/spark/unsafe/types/CalendarInterval.java [hunk 1]
```diff
@@ -29,7 +29,7 @@
 /**
  * The internal representation of interval type.
  */
-public final class CalendarInterval implements Serializable, Comparable<CalendarInterval> {
+public final class CalendarInterval implements Serializable {
```
Comment (cloud-fan) [inline common/unsafe/src/main/java/org/apache/spark/unsafe/types/CalendarInterval.java]: there is a `MAX_VALUE` and `MIN_VALUE` in this file, shall we remove them as well?

File: sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/ColumnStats.scala [hunk 2]
```diff
@@ -296,14 +296,11 @@ private[columnar] final class BinaryColumnStats extends ColumnStats {
 }
 
 private[columnar] final class IntervalColumnStats extends ColumnStats {
-  protected var upper: CalendarInterval = CalendarInterval.MIN_VALUE
-  protected var lower: CalendarInterval = CalendarInterval.MAX_VALUE
+  protected var upper: CalendarInterval = null
+  protected var lower: CalendarInterval = null
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/ColumnStats.scala]: can we follow `BinaryColumnStats` and don't define the `upper` and `lower` variable?

**Comment by cloud-fan** [comment]
shall we also revert https://github.com/apache/spark/pull/26681 ?

File: common/unsafe/src/main/java/org/apache/spark/unsafe/types/CalendarInterval.java (hunk 1 above)
Comment (yaooqinn) [reply common/unsafe/src/main/java/org/apache/spark/unsafe/types/CalendarInterval.java]: yes, remove them too

File: sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/ColumnStats.scala (hunk 2 above)
Comment (yaooqinn) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/ColumnStats.scala]: yes

**Comment by yaooqinn** [comment]
> shall we also revert #26681 ?

thanks, I missed this one

**Review by cloud-fan — APPROVED** [review]

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
Please be clear for the reverting in the PR description, @yaooqinn .
If SPARK-30048 is clearly reverted by this PR, could you update SPARK-30048 JIRA, @cloud-fan ?

**Comment by dongjoon-hyun** [comment]
Also, cc @maropu

**Comment by cloud-fan** [comment]
I've updated SPARK-29679 and SPARK-30048 as won't do and mention they have been reverted.

Thanks, merging to master!

**Comment by dongjoon-hyun** [comment]
Thank you for the clarification, @cloud-fan and @yaooqinn .

**Comment by maropu** [comment]
Checked; late LGTM.

**Pull request closed without merging**